realm/realm-library/src/main/cpp/io_realm_internal_OsResults.cpp

```diff
@@ -323,6 +324,91 @@ JNIEXPORT jboolean JNICALL Java_io_realm_internal_OsResults_nativeDeleteFirst(JN
     return JNI_FALSE;
 }
 
+static inline void update_objects(JNIEnv* env, jlong results_ptr, jstring& j_field_name, JavaValue& value) {
+    try {
```

Comment:
I think you need to move the `try`/`CATCH_STD()` to _all_ the actual JNI functions.